packages/protocol/contracts/common/FeeCurrencyWhitelist.sol

```diff
@@ -4,26 +4,37 @@ import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 
 import "./interfaces/IFeeCurrencyWhitelist.sol";
 
-import "../common/Initializable.sol";
+import "./InitializableV2.sol";
 
+import "./UsingRegistry.sol";
+import "../stability/interfaces/ISortedOracles.sol";
 /**
  * @title Holds a whitelist of the ERC20+ tokens that can be used to pay for gas
  */
-contract FeeCurrencyWhitelist is IFeeCurrencyWhitelist, Ownable, Initializable {
+contract FeeCurrencyWhitelist is IFeeCurrencyWhitelist, Ownable, InitializableV2, UsingRegistry {
   address[] public whitelist;
 
+  constructor(bool test) public InitializableV2(test) {}
+
   /**
    * @notice Used in place of the constructor to allow the contract to be upgradable via proxy.
+   * @param registryAddress Address of the Registry contract.
    */
-  function initialize() external initializer {
+  function initialize(address registryAddress) external initializer {
     _transferOwnership(msg.sender);
+    setRegistry(registryAddress);
```

Comment:
By introducing new storage, we imply that `FeeCurrencyWhitelist` will have a new storage contract deployed and initialized.

See https://docs.celo.org/community/release-process/smart-contracts#versioning

If we were to redeploy `FeeCurrencyWhitelist`, we would also need to migrate the existing whitelist to the new address atomically in the `Governance` proposal. It was an oversight not to deploy this contract with `UsingRegistry` in the first place, but I think the upgrade complexity is not worth the syntactical benefits we gain. 

My recommendation would be to give `FeeCurrencyWhitelist` special knowledge of the (likely) indefinitely static registry address `0x0..ce10` in its bytecode. 

cc @nambrot